octavia-cli/octavia_cli/list/listings.py

```diff
@@ -0,0 +1,111 @@
+        pass
+
+    @property
+    @abc.abstractmethod
+    def list_field_in_response(
+        self,
+    ) -> str:  # pragma: no cover
+        pass
+
+    @property
+    @abc.abstractmethod
+    def list_function_name(
+        self,
+    ) -> str:  # pragma: no cover
+        pass
+
+    @property
+    def _list_fn(self):
+        return getattr(self.api, self.list_function_name)
+
+    @property
+    def list_function_kwargs(self) -> dict:
+        return {}
+
+    def __init__(self, api_client: airbyte_api_client.ApiClient):
+        self.api_instance = self.api(api_client)
+
+    def _parse_response(self, api_response) -> List[List[str]]:
+        items = [[item[field] for field in self.fields_to_display] for item in api_response[self.list_field_in_response]]
+        return items
+
+    def get_listing(self) -> List[List[str]]:
+        api_response = self._list_fn(self.api_instance, **self.list_function_kwargs, **self.COMMON_LIST_FUNCTION_KWARGS)
+        return self._parse_response(api_response)
+
+    def __repr__(self):
+        items = [formatting.format_column_names(self.fields_to_display)] + self.get_listing()
+        return formatting.display_as_table(items)
+
+
+class SourceConnectorsDefinitions(BaseListing):
+    api = source_definition_api.SourceDefinitionApi
+    fields_to_display = ["name", "dockerRepository", "dockerImageTag", "sourceDefinitionId"]
+    list_field_in_response = "source_definitions"
+    list_function_name = "list_latest_source_definitions"
+
+
+class DestinationConnectorsDefinitions(BaseListing):
+    api = destination_definition_api.DestinationDefinitionApi
+    fields_to_display = ["name", "dockerRepository", "dockerImageTag", "destinationDefinitionId"]
+    list_field_in_response = "destination_definitions"
+    list_function_name = "list_latest_destination_definitions"
+
+
+class WorkspaceListing(BaseListing, abc.ABC):
+    def __init__(self, api_client: airbyte_api_client.ApiClient, workspace_id: str):
+        self.workspace_id = workspace_id
+        super().__init__(api_client)
+
+    @property
+    def list_function_kwargs(self) -> dict:
+        return {"workspace_id_request_body": WorkspaceIdRequestBody(workspace_id=self.workspace_id)}
+
+
+class Sources(WorkspaceListing):
+    api = source_api.SourceApi
+    fields_to_display = ["name", "sourceName", "sourceId"]
+    list_field_in_response = "sources"
+    list_function_name = "list_sources_for_workspace"
+
+
+class Destinations(WorkspaceListing):
+    api = destination_api.DestinationApi
+    fields_to_display = ["name", "destinationName", "destinationId"]
+    list_field_in_response = "destinations"
+    list_function_name = "list_destinations_for_workspace"
+
+
+class Connections(WorkspaceListing):
+    api = connection_api.ConnectionApi
+    fields_to_display = ["name", "connectionId", "status", "sourceId", "destinationId"]
+    list_field_in_response = "connections"
+    list_function_name = "list_connections_for_workspace"
```

Comment:
The fact that these inherit from a WorkspaceListing made me think: would it make sense to group these commands under a `workspace` keyword? I.e. the commands would look like
```
octavia list workspace sources
octavia list workspace destinations
octavia list workspace connections
octavia list connectors sources
octavia list connectors destinations
```
Do you think this would make it more clear what the difference between these commands are to the user? Or would we just end up with a ton of `workspace` commands, making the commands unnecessarily long?

I'll leave it up to you to make the final decision here; I don't feel super strongly either way, but just thought I would offer this suggestion in case you liked it

Reply:
I like this, looks tidier and will keep the same command "depth" for `list` subcommands.